### ISSUE TITLE
chore: Set up CI

### ISSUE THREAD

**Issue by moyix** [moyix/fauxpilot#98]
Now that we're getting a lot more contributions (yay!), reviewing and testing PRs by hand is starting to get unmanageable. So we need to set up some basic CI. This will also help with the problem where I keep forgetting to update and push the container images when something in them gets updated!

Steps:

- [ ] Make a GH Actions workflow that pulls in the necessary repos, builds the three containers we use (copilot_proxy, triton_with_ft, and model_converter), and then pushes them to Docker Hub, so that I don't have to remember to do it by hand.
- [ ] Create some very basic unit tests for the proxy
- [ ] Maybe make a mock inference server that the proxy can talk to to verify that its completion endpoint is working?
- [ ] Testing the Triton/FasterTransformer part is probably the hardest here since we don't have any GPU runners (Yet! Growth mindset!), but maybe the Python backend in #73 will help with that since its models can run on the CPU (slowly)

**Comment by leemgs** [comment]
I like this issue. CI/CD is very important for software maintenance and quality maintenance. Although we do a lot of review, merged PRs often become another potential bug.

In my experience, the simplest way is to utilize GH Actions. This is because, in the case of open source projects, many features are available for free. Meanwhile, like FauxPilot Server, CI operations that require GPU resources and system environment settings often require their own server. In this case, we can use [Jenkins](https://github.com/jenkinsci/jenkins). But this software is heavy. So I made and operated [TAOS-CI](https://github.com/nnstreamer/TAOS-CI) lightly.

In conclusion, In order to prevent unexpected bugs in advance while handling code inflow from the FauxPilot repository quickly
I think it's time to think about CI/CD. Using my spare time, I will also contribute to creating related PRs.
